tgctoolbox/settings.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import os
+
+import yaml
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+BASE_FILES = [
+    os.path.join("configuration", "base.yaml"),
+    os.path.join("configuration", "local.yaml"),
+    os.path.join("configuration", "production.yaml"),
+]
+
+
+class Settings:

```

Comment:
please, refer to:

https://refactoring.guru/design-patterns/singleton/python/example#example-0

as you can see there are some tricks which makes it more easier

Reply:
I literally do what is defined here. Code from the link looks like:

```python
def __call__(cls, *args, **kwargs):
        """
        Possible changes to the value of the `__init__` argument do not affect
        the returned instance.
        """
        if cls not in cls._instances:
            instance = super().__call__(*args, **kwargs)
            cls._instances[cls] = instance
        return cls._instances[cls]
```

Our code:

```python
def __new__(cls, *args, **kwargs):
        if cls._instance is None:
            cls._instance = super().__new__(cls)
        return cls._instance
```

This does the same, just without declaring the meta class.

Reply:
Oh well then, that was my skill issue